add dialog for voting

aguycalled opened the pull request:
Adds a dialog for managing voting of the community fund.

marcus290 committed:
combined cfund test helper functions: activate_cfund, start_new_cycle, end_cycle, slow_gen

marcus290 committed:
simplified setup-network using existing code

marcus290 committed:
removed duplicate code from rawtx tests and minor linting fixes

Commit:
add dialog for voting

matt-auckland commented on src/qt/forms/moc_cfund_voting.cpp:
```diff
@@ -0,0 +1,145 @@
+/****************************************************************************
+** Meta object code from reading C++ file 'cfund_voting.h'
```
Is this normal?

matt-auckland commented on src/qt/forms/moc_cfund_proposals.cpp:
```diff
@@ -0,0 +1,145 @@
+/****************************************************************************
+** Meta object code from reading C++ file 'cfund_proposals.h'
+**
+**
```
is this normal?

Commit:
rm file with old name

matt-auckland reviewed (commented):
Code looks good, will approve after testing

Commit:
update dialog voting cfund

Commit:
remove old file

Commit:
show warning when new cfund entries

matt-auckland commented on src/qt/forms/overviewpage.ui:
```diff
@@ -511,6 +486,26 @@
+          <string notr="true">color:#cc2222</string>
+         </property>
+         <property name="text">
+          <string>Wallet syncing, the balance won't be accurate until the process finishes</string>
```
```suggestion
          <string>This wallet is currently syncing. Your balance may not be accurate until it has completed</string>
```
Slight wording tweak

matt-auckland commented:
Shouldn't this be a PR into master? If we want to keep our change atomic like we discussed the other day?

matt-auckland commented on src/qt/navcoingui.cpp:
```diff
@@ -420,6 +426,8 @@ void NavCoinGUI::createActions()
     optionsAction->setStatusTip(tr("Modify configuration options for %1").arg(tr(PACKAGE_NAME)));
     optionsAction->setMenuRole(QAction::PreferencesRole);
     optionsAction->setEnabled(false);
+    cfundProposalsAction = new QAction(tr("Voting of Proposals"), this);
```
```suggestion
    cfundProposalsAction = new QAction(tr("Vote for Proposals"), this);
```

matt-auckland commented on src/qt/navcoingui.cpp:
```diff
@@ -420,6 +426,8 @@ void NavCoinGUI::createActions()
     optionsAction->setMenuRole(QAction::PreferencesRole);
     optionsAction->setEnabled(false);
+    cfundProposalsAction = new QAction(tr("Voting of Proposals"), this);
+    cfundPaymentRequestsAction = new QAction(tr("Voting of Payment Requests"), this);
```
```suggestion
    cfundPaymentRequestsAction = new QAction(tr("Vote for Payment Requests"), this);
```

matt-auckland commented on src/qt/navcoingui.cpp:
```diff
@@ -1766,6 +1797,62 @@ void NavCoinGUI::updateStakingStatus()
+                    }
+                }
+            }
+            if (fFound && !this->fNotShowAgain && (this->lastDialogShown + (60*60*24)) < GetTimeNow()) {
```
```suggestion
            if (fFound && !this->fDon'tShowAgain && (this->lastDialogShown + (60*60*24)) < GetTimeNow()) {
```

matt-auckland commented on src/qt/navcoingui.cpp:
```diff
@@ -1766,6 +1797,62 @@ void NavCoinGUI::updateStakingStatus()
+                msgbox.exec();
+
+                if(cb->isChecked()) {
+                    this->fNotShowAgain = true;
```
```suggestion
                    this->fDon'tShowAgain = true;
```

matt-auckland commented on src/qt/navcoingui.cpp:
```diff
@@ -1766,6 +1797,62 @@ void NavCoinGUI::updateStakingStatus()
+                if(cb->isChecked()) {
+                    this->fNotShowAgain = true;
+                } else {
+                    this->fNotShowAgain = false;
```
```suggestion
                    this->fDon'tShowAgain = false;
```

matt-auckland commented on src/qt/navcoingui.h:
```diff
@@ -143,6 +145,9 @@ class NavCoinGUI : public QMainWindow
     int prevBlocks;
     int spinnerFrame;
 
+    bool fNotShowAgain;
```
```suggestion
    bool fDon'tShowAgain;
```

matt-auckland commented on src/qt/navcoingui.cpp:
```diff
@@ -145,6 +149,8 @@ NavCoinGUI::NavCoinGUI(const PlatformStyle *platformStyle, const NetworkStyle *n
     helpMessageDialog(0),
     prevBlocks(0),
     spinnerFrame(0),
+    fNotShowAgain(false),
```
```suggestion
    fDon'tShowAgain(false),
```

just tweaking the wording of this variable a bit

matt-auckland commented on src/qt/navcoingui.cpp:
```diff
@@ -277,7 +283,7 @@ NavCoinGUI::NavCoinGUI(const PlatformStyle *platformStyle, const NetworkStyle *n
     QTimer *timerStakingIcon = new QTimer(labelStakingIcon);
     connect(timerStakingIcon, SIGNAL(timeout()), this, SLOT(updateStakingStatus()));
-    timerStakingIcon->start(150 * 1000);
+    timerStakingIcon->start(45 * 1000);
```
Does this mean the wallet checks if we are staking or not even 45 seconds?

aguycalled replied:
yes

matt-auckland commented on src/qt/navcoingui.cpp:
```diff
@@ -1766,6 +1797,62 @@ void NavCoinGUI::updateStakingStatus()
+                }
+            }
+            if (fFound && !this->fNotShowAgain && (this->lastDialogShown + (60*60*24)) < GetTimeNow()) {
+                QCheckBox *cb = new QCheckBox("Do not show this message again during this session.");
```
```suggestion
                QCheckBox *cb = new QCheckBox("Don't show this notification again until wallet is restarted.");
```

matt-auckland committed:
Made several tweaks to the cfund gui. Fixed some spelling and wording. Added a new title to the dialog so we can see easily if we are looking at proposals or payment requests. Moved some buttons around.

matt-auckland committed:
Made several tweaks to the cfund gui. Fixed some spelling and wording. Added a new title to the dialog so we can see easily if we are looking at proposals or payment requests. Improved button placement. Selection clears after voting and page swapping. buttons are disabled unless you select something

matt-auckland committed:
fixed syntax error. added methods to header file. swapped string -> QString

matt-auckland committed:
buttons should now disable when you select an ite from their list. i.e. select an item from the no list, the vote no button should disable

matt-auckland committed:
fixed typo

matt-auckland committed:
Swap 'view proposal details' button text when we change views

matt-auckland committed:
Tweaks to the cfund gui (#43)

* Made several tweaks to the cfund gui. Fixed some spelling and wording. Added a new title to the dialog so we can see easily if we are looking at proposals or payment requests. Improved button placement. Selection clears after voting and page swapping. buttons are disabled unless you select something

* fixed syntax error. added methods to header file. swapped string -> QString

* buttons should now disable when you select an ite from their list. i.e. select an item from the no list, the vote no button should disable

* fixed typo

* Swap 'view proposal details' button text when we change views

matt-auckland committed:
Tweaked list to prevent weird autohighlighting bug

matt-auckland committed:
merged

matt-auckland committed:
merged

matt-auckland committed:
merged cfund-gui

matt-auckland committed:
removed extra spacer

matt-auckland committed:
Update src/qt/forms/cfund_voting.cpp

removed whitespace

matt-auckland committed:
tweaked wording of cfund alert box

matt-auckland committed:
Merge branch 'cfund-gui-tweaks' of github.com:matt-auckland/navcoin-core into cfund-gui-tweaks

marcus290 committed:
Merge remote-tracking branch 'upstream/master'

marcus290 committed:
updates bootstrap link

matt-auckland committed:
Merge pull request #44 from matt-auckland/cfund-gui-tweaks

Cfund gui tweaks

matt-auckland committed:
Merge branch 'cold-staking' into cfund-gui

matt-auckland committed:
Added draft release notes

marcus290 committed:
Merge remote-tracking branch 'upstream/master'

marcus290 committed:
update release notes

proletesseract committed:
Merge pull request #338 from marcus290/master

Wallet GUI: Updates bootstrap link in wallet

matt-auckland committed:
Tweaked draft release notes

Commit:
Merge branch 'cold-staking' into cfund-gui

Commit:
Cold staking (#249)

* stake weight counts cold staking balance

* show colstaking output values in gui

* overview layout

* overview layout move to staking section

* receive layout

* fix top menu layout

* unused var

* set coinstake version before signing

* resign coinstake tx

* error takes string as arg

* error must return false

* inherit return value

* extract pub key from input scripsig when cold staking

* gui cold staking wizard

* fix layout

* fix layout

* restrict getcoldstakingaddress only after actv

* use chainactive instead of pindexbestheader

* add logging

* correctly extract pubkey from coldstaking script

* new regtest genesis block

* IsPayToPublicKey()

* addressindex and unspentindex track p2pk (staking) outputs

* use address type 1 for p2pk

* use extract destination for address/unspent indexes

* track cold staking addresses in index

* GetStakingKeyID / GetSpendingKeyID

Add new class methods to obtain the staking and spending key ids from a cold staking address

* ui checks if spends coldstaking out

when sending a transaction, the ui will check whether any of the outputs is a cold staking script. in that case, it will warn the user about the coins going to a normal change address if he didnt specify a cold staking address as custom change destination

* fix

* update strings

* add IsColdStake() to CScript

* parse OP_COINSTAKE opcode

* Solver() extracts addresses from cold staking tx

* read key from coldstaking script when verifying block signature

* sign coldstaking input

* remove cnavcoinaddress

* add key_io

* revert

* fix extractdestinations

* required signatures

* show cold staking balance

* get available stakable credit

* get available stakable credit

* count cold staking balance when creating coinstake

* show canspend / canstake in listtransactions

* fix total stakable amount

* list transactions canspend/stake

* fix order of the public keys on raw decoded address

* cold staking balance in ui

* show colstaking output values in gui

* overview layout

* overview layout move to staking section

* fix top menu layout

* extract pub key from input scripsig when cold staking

* gui cold staking wizard

* add logging

* remove duplicate case

* Remove duplicated case

* Remove wrong duplicated check

* do not enforce random string for preq v2

* adjust overview layout

* fix condition order

* extend log

* move proposal/preq check to contextualcheckblock

* revert

* rebase cold staking branch

* testnet coldstaking versionbit

* fix version bit

* check versionbit testnet

* sync branch

* restart testnet / increase pow last block

* add log

* cfund change logic for accepting props/preq

* add consensus info to cfundstats

* do not count rejected preqs when creating a new one

* added cold staking to regtest (#31)

* combined cfund test helper functions: activate_cfund, start_new_cycle, end_cycle, slow_gen

* simplified setup-network using existing code

* removed duplicate code from rawtx tests and minor linting fixes

* adds cold staking to regtest

* added cold staking to regtest

* new testnet genesis

* a coldstakingaddress can be constructued only from two normal addresses

* bug fix for getcoldstakingaddress (#32)

* forces user to use different staking and spending addresses for getcoldstakingaddress

* updated the cold-staking wizard so you cannot enter the same address as both args

* add coldstaking to regtest

* admit holding two priv keys of a cstaking script

* Merge from Cfund cold staking (#33)

* do not enforce random string for preq v2

* adjust overview layout

* fix condition order

* extend log

* move proposal/preq check to contextualcheckblock

* revert

* restart testnet / increase pow last block

* add log

* cfund change logic for accepting props/preq

* add consensus info to cfundstats

* do not count rejected preqs when creating a new one

* new testnet genesis

* a coldstakingaddress can be constructued only from two normal addresses

* bug fix for getcoldstakingaddress (#32)

* forces user to use different staking and spending addresses for getcoldstakingaddress

* updated the cold-staking wizard so you cannot enter the same address as both args

* add coldstaking to regtest

* admit holding two priv keys of a cstaking script

* FIxed wizard text (#35)

* fix overview page

* fix coldstaking wizard

* show both staking and spending pubkeys

* fix cold staking wizard

* combine signatures cold staking

* Moved over RPC tests from other branch

* Updated gitignore

* ported fix for zlib compile on mac from bitcoin https://github.com/bitcoin/bitcoin/pull/9973

* tided tests up slightly

* First spending test

* Largely completed tests for coldstaking where we hold the spending address

* started work on coldstaing tests where we hold the staking address

* fixed accidental var reasignment in test

* Added a staking test

* added some debugging to test

* added missing brackets to function call

* added a loop that waits until we stake

* fixed syntax error

* fixed some more syntax errors

* replaced missing bracket

* Updated test so we use getblockheight instead of getblockchaininfo and only set up 1 node instaed of 2

* we now mine blocks so our coins mature for stkaing

* now we check if the cold staking utxo is used for staking

* adding in block generation so we confirm our tx to the cold stkaing address

* ading more logs for debugging

* more debugging

* syncing blocks between nodes

* Rejigged node setup, hopefully gets syncing working

* get the utxo a different way

* fixed syntax error

* New method to get utxo

* fixes some syntax errors

* fixes other syntax errors

* We check if the uxto is spent via staking

* bump version 4.5.0

* added some more staing address tests

* added ore logging

* updated arg name in send raw tx function, os it's more verbose

* Updated function so we throw a custom exception we can explicitely catch

* fixed exception throwing

* fixed syntax err

* fixing more syntax errors

* playing around with exception catching

* fixed typo

* Removed some logging, unused variables and commented out some printing as part of test clean up

* added syncing to test

* more test fixes

* removed syncing

* removed syncing

* wrapped operation in rbackets

* parse  numbers asfloats

* removed back slahses

* fixed syntax error

* we generate our blocks on node 0 now and just remove the block rewards value from our balance calculatins

* added getcoldstaking error message

* updated rpc test for new help text

* adds modified coldstaking client testing doc

* format fixes

* format fixes

* Moved over RPC tests from other branch

* Updated gitignore

* ported fix for zlib compile on mac from bitcoin https://github.com/bitcoin/bitcoin/pull/9973

* tided tests up slightly

* First spending test

* Largely completed tests for coldstaking where we hold the spending address

* started work on coldstaing tests where we hold the staking address

* fixed accidental var reasignment in test

* Added a staking test

* added some debugging to test

* added missing brackets to function call

* added a loop that waits until we stake

* fixed syntax error

* fixed some more syntax errors

* replaced missing bracket

* Updated test so we use getblockheight instead of getblockchaininfo and only set up 1 node instaed of 2

* we now mine blocks so our coins mature for stkaing

* now we check if the cold staking utxo is used for staking

* adding in block generation so we confirm our tx to the cold stkaing address

* ading more logs for debugging

* more debugging

* syncing blocks between nodes

* Rejigged node setup, hopefully gets syncing working

* get the utxo a different way

* fixed syntax error

* New method to get utxo

* fixes some syntax errors

* fixes other syntax errors

* We check if the uxto is spent via staking

* added some more staing address tests

* added ore logging

* updated arg name in send raw tx function, os it's more verbose

* Updated function so we throw a custom exception we can explicitely catch

* fixed exception throwing

* fixed syntax err

* fixing more syntax errors

* playing around with exception catching

* fixed typo

* Static reward RC (#328)

* static rewards added

* initial test and test util added

* added new test to rpc test file

* test updated

* adding sleep print

* removed extra bracket

* updated perms.

* cleanup test

* checking before the soft fork

* creating a few blocks to start

* removing pre check

* Removed some logging, unused variables and commented out some printing as part of test clean up

* added syncing to test

* more test fixes

* removed syncing

* removed syncing

* wrapped operation in rbackets

* parse  numbers asfloats

* removed back slahses

* fixed syntax error

* we generate our blocks on node 0 now and just remove the block rewards value from our balance calculatins

* added getcoldstaking error message

* updated rpc test for new help text

* adds modified coldstaking client testing doc

* format fixes

* format fixes

* fix

* added spending-from-coldstaking test

* updated rpc-tests.py

* removed extra file, updated coldstaking tests

* Update qa/pull-tester/rpc-tests.py

delete reference to deleted file

* Updated release notes with a draft of the cold staking notes

* added notes about Marcus's testing

* fix word

Commit:
Merge remote-tracking branch 'upstream/master' into cfund-gui

matt-auckland committed:
Update doc/release-notes/release-notes-4.5.0.md

Co-Authored-By: aguycalled <alex@encrypt-s.com>

Commit:
Update release-notes-4.5.0.md (#339)

Commit:
Merge branch 'master' into cfund-gui

Commit:
add sleep to spending test

Commit:
Merge branch 'cfund-gui' of https://github.com/aguycalled/navcoin-core into cfund-gui

Commit:
coldstaking spending test fix

Commit:
increase fee spending cs test
